syscall/syscall.csv

```diff
@@ -83,14 +84,14 @@
 "pselect","sys/select.h","","int","int","FAR fd_set *","FAR fd_set *","FAR fd_set *","FAR const struct timespec *","FAR const sigset_t *"
 "pthread_cancel","pthread.h","!defined(CONFIG_DISABLE_PTHREAD)","int","pthread_t"
 "pthread_cleanup_pop","pthread.h","defined(CONFIG_PTHREAD_CLEANUP)","void","int"
+"pthread_cleanup_poplist","pthread.h","defined(CONFIG_PTHREAD_CLEANUP)","int","struct pthread_cleanup_s *"
 "pthread_cleanup_push","pthread.h","defined(CONFIG_PTHREAD_CLEANUP)","void","pthread_cleanup_t","FAR void *"
```

Comment:
pthread_cleanup_poplist currently uses data in the TCB to save the cleanup info.  That is a problem.

That data needs to be move to TLS and pthread_cleanup_poplist() needs to be moved to libs/libc/pthread.  This should all be done in user space.

Reply:
Got it